### ISSUE TITLE
Loader for Che - Preliminary Design Exploration

### ISSUE THREAD

**Issue by catrobson** [openshiftio/openshift.io#2875]
To Do:
- [x] Post sketches done thus far (as of 19 April)
- [x] Meet with @slemeur 's team to review, possibly brainstorm on new concepts
- [x] Take feedback and continue new work

We have some change in the way we are going to load the IDE.
Until now, the when user's workspace was started, we were displaying the IDE and in the IDE showing the loading of all the workspace's containers. Once all workspace's containers were ready, we let the user being able to interact with the file

With the new IDE that we are introducing, that will be a little bit different:
1. The IDE is not be available upfront
2. We have an intermediate step while the IDE is getting assembled (depending on plugins selection)
3. We then start the IDE
4. Once the IDE is ready, we display it and make it "available" for editing
5. Workspace's containers are loaded in parallel (but that does not prevent the user to start interacting with its file for example) 

I would think that this gives us an opportunity to make a really creative loader.  know that this would be something low regarding your other priorities, but do you think it is something your team could help us to build?

- [Meeting notes from 27 Feb 2018 with @ashumilova and @slemeur](https://docs.google.com/document/d/16-4m2nZMaxEAOrquVCAE6FmR_gZpLKtCJ4LqA-NgQpY/edit?usp=sharing)
- Link to [loader video](https://youtu.be/ra8MlcddbBs) of IDE loading
- [Reference/inspiration images](https://drive.google.com/open?id=19m7jl7X6Pf2t9ixwLlrWShQt1DKKfTBj)
- [Links to other inspiration imagery](https://docs.google.com/document/d/1K9FcIOHxL0G4lUdgeTSrm5zPSrwN8NS8TiV98Isxo3w/edit?usp=sharing)

**Comment by bdellasc** [comment]
- Met this morning with @slemeur @ashumilova @catrobson and Vitaliy to walk through sketches and ideas
- Added some additional inspirational imagery to [this doc](https://docs.google.com/document/d/1K9FcIOHxL0G4lUdgeTSrm5zPSrwN8NS8TiV98Isxo3w/edit?usp=sharing)
- Concepts to continue with/use as inspiration: 2, 4, 5, 9, 10
- Regroup next week to check in on progress made

**Comment by bdellasc** [comment]
- Created another concept, but in digital form
- Presented the concept in demo
- Hunted some more for a good WYSIWYG SVG animation tool, and seemed to have found one
- Brought digital concept into the tool and began playing around with the animation ideas I had
- Have a couple more ideas that I want to flesh out as "vignettes" on style and motion to get some reactions from @slemeur 's team. There's more work to do before I get there.

**Comment by slemeur** [comment]
Sounds good. Thanks @bdellasc !

**Comment by bdellasc** [comment]
Because of the amount of time the exploration is taking, I'm going to spin off the final animation work into a separate story. I'd like to finish this week with preference down to one or two general directions for the final animation.

**Comment by bdellasc** [comment]
Met with @slemeur 's team late last week. We agreed they have some investigation before final animation work can commence. [Notes added to this document](https://docs.google.com/document/d/1UBudUth8GGfPRlyw9jkke3bGbZNgjCNx6ZUFwK3yoFc/edit?usp=sharing).
